File: FastPdfKit-master/Reader/OverlayManager.m
//
//  OverlayManager.m
//  FPKSimpleApp
//
//  Created by Maahi on 06/03/18.
//

#import "OverlayManager.h"
#import <FastPdfKit/FPKGlyphBox.h>
#import "MFDocumentManager.h"
//#import "Drawable.h"
@implementation OverlayManager
@synthesize searchKeyword;
@synthesize documentManager;
@synthesize glyphBoxes;
@synthesize selectedArea;

-(NSArray *)documentViewController:(MFDocumentViewController *)dvc drawablesForPage:(NSUInteger)page {
    
    if (self.searchKeyword!=nil) {
        NSArray *fsgsg=[documentManager glyphBoxesForPage:page];
        return [self.documentManager searchResultOnPage:page forSearchTerms:self.searchKeyword ignoreCase:YES];
    }else{
        NSMutableArray *drawables = [[NSMutableArray alloc]init];
        for (NSValue * tmpValue in self.selectedArea) {
            UIView * selectedView = [[UIView alloc] initWithFrame:[tmpValue CGRectValue]];
            [drawables addObject:selectedView];
        }
        return drawables;
        
    }
    return nil;
}



//////


- (void) startSelectedArea:(NSInteger)page {
    self.glyphBoxes = [self.documentManager glyphBoxesForPage:page];
}


- (BOOL) calculateSelectedArea:(CGPoint) point page:(NSInteger)page {
    CGRect tmpRect = CGRectZero;
    for (FPKGlyphBox *glyphBox in self.glyphBoxes) {
        CGRect glyphBoxRect = glyphBox.box;
        if (CGRectContainsPoint(glyphBoxRect, point)) {
            tmpRect = glyphBoxRect;
        }
    }
    if (!CGRectIsEmpty(tmpRect)) {
        CGFloat valueOriginY = tmpRect.origin.y;
        
        for (FPKGlyphBox * glyphBox in self.glyphBoxes) {
            CGRect glyphBoxRect = glyphBox.box;
            CGFloat originY = glyphBoxRect.origin.y;
            if (valueOriginY == originY) {
                tmpRect = CGRectUnion(glyphBoxRect, tmpRect);
            }
        }
        
    }
    
    
    if (!CGRectIsEmpty(tmpRect)) {
        BOOL setted = NO;
        for (NSUInteger cpt = 0; cpt < [self.selectedArea count]; cpt++) {
            NSValue * tmpValue = [self.selectedArea objectAtIndex:cpt];
            CGRect rectValue = [tmpValue CGRectValue];
            CGFloat valueOriginY = rectValue.origin.y;
            CGFloat originY = tmpRect.origin.y;
            if (valueOriginY == originY) {
                setted = YES;
            }
        }
        if (!setted) {
            [self.selectedArea addObject:[NSValue valueWithCGRect:tmpRect]];
        }
    }
    
    return YES;
}


@end

File: MNA/FastPDF/OverlayManager.swift
//
//  OverlayManager.swift
//  MNA
//
//  Created by Maahi on 20/01/18.
//  Copyright © 2018 KTeck. All rights reserved.
//

import UIKit

class OverlayManager: NSObject ,MFDocumentOverlayDataSource{

    var searchKeyword = ""
    var documentManager:MFDocumentManager = MFDocumentManager()
    var glyphBoxes:NSMutableArray =  NSMutableArray()
    var selectedArea:NSMutableArray = NSMutableArray()
    
    
    
    func documentViewController(_ dvc:MFDocumentViewController , drawablesForPage:Int) -> NSArray{
        
        if searchKeyword != ""{
           
            //let   fsgsg:NSArray =  documentManager.glyphBoxes(forPage: UInt(drawablesForPage)) as! NSArray
            return  documentManager.searchResult(onPage: UInt(drawablesForPage), forSearchTerms: searchKeyword, ignoreCase: true)! as NSArray
           // (onPage: drawablesForPage, forSearchTerms: searchKeyword, ign)
        
        
        }else{
            
            var drawables:NSMutableArray = NSMutableArray()
            for tmpValue in selectedArea {
                let tVal  = tmpValue as! UIView
                let selectedView:UIView = UIView()
                selectedView.frame = tVal.frame
                drawables.add(selectedView)
                
            }
            
            return drawables
            
        }
        
      //  return NSArray()
    }
    
    
    
    
    func startSelectedArea(_ page:Int) {
        glyphBoxes = documentManager.glyphBoxes(forPage: UInt(page)) as! NSMutableArray
    }
    
    
    
    
    
    func calculateSelectedArea(_ point:CGPoint, page:Int) -> Bool{
        
        var tmpRect:CGRect = CGRect()
        for tmpValue in glyphBoxes {
            let glyphBox = tmpValue as! FPKGlyphBox
            let glyphBoxRect :CGRect =  glyphBox.box
            
            if glyphBoxRect.contains(point){
                tmpRect = glyphBoxRect
            }
            
        }
        
        if  !self.CheckEmpity(tmpRect) {
            let valueOriginY:CGFloat = tmpRect.origin.y
            
            for tmpValue in glyphBoxes {
                let glyphBox = tmpValue as! FPKGlyphBox
                let glyphBoxRect :CGRect =  glyphBox.box
                
                let originY:CGFloat = glyphBoxRect.origin.y
                if valueOriginY == originY {
                    tmpRect = glyphBoxRect.union(tmpRect);
                }
            }
            
        }
        
        
        
        
        
        
        
        return true
    }
    
    
    func CheckEmpity(_ frm:CGRect) ->Bool{

        let empity:CGRect? = nil
        if frm == empity{
            return true
        }else{
            return false
        }
        
    }
    
}
